Dockerfile

```diff
@@ -1,26 +1,37 @@
 FROM node:14-slim AS make-bundle
+# Create working directory as env variable
 ENV WEB_APP=/home/app/webapp
+# Move into working directory
 WORKDIR $WEB_APP
+# Copy app from local into the container 
 COPY . $WEB_APP
+# Install relevant packages and bundle JS files with webpack 
 RUN npm install && \
     npm run webpack 
 
-
-
-FROM python:3.7  
-ENV WEB_APP=/home/app/webapp   
-WORKDIR $WEB_APP  
-RUN pip install --upgrade pip  
-ENV PYTHONDONTWRITEBYTECODE 1 \
-    PYTHONUNBUFFERED 1  
+FROM python:3.7 
+# Create working directory as env variable
+# PYTHONDONTWRITEBYTECODE=1 This prevents Python from writing out pyc files 
+# PYTHONUNBUFFERED=1 This keeps Python from buffering stdin/stdout
+ENV WEB_APP=/home/app/webapp PYTHONDONTWRITEBYTECODE=1 PYTHONUNBUFFERED=1 
+# Move into working directory 
+WORKDIR $WEB_APP 
+# Make sure pip is up to date 
+RUN pip install --upgrade pip
+# Copy and install requirements  
 COPY ./requirements.txt $WEB_APP
-RUN pip install -r requirements.txt 
+RUN pip install -r requirements.txt
+# Copy app from local into the container  
 COPY . $WEB_APP
+# Copy the bundled JS from previous build
 COPY --from=make-bundle /home/app/webapp/static/dist/bundle.js $WEB_APP/static/dist/bundle.js 
+# Create a user give it ownership of the app and give only the user permission to change the app
```

Comment:
this actually makes sure app is running with user (limited) privilages instead of root. in case of a hack to your python code via some code injection or whatever, the hacker will be limited to run stuff in the container as she'll be running under the user priviliges